lua/nvim-treesitter/parser_configs.lua

```diff
@@ -0,0 +1,358 @@
+  install_info = {
+    url = "https://github.com/ikatyang/tree-sitter-toml",
+    files = { "src/parser.c", "src/scanner.c" },
+  },
+  maintainers = {"@tk-shirasaka"},
+}
+
+list.vue = {
+  install_info = {
+    url = "https://github.com/ikatyang/tree-sitter-vue",
+    files = { "src/parser.c", "src/scanner.cc" },
+  }
+}
+
+list.elm = {
+  install_info = {
+    url = "https://github.com/razzeee/tree-sitter-elm",
+    files = { "src/parser.c", "src/scanner.cc" },
+  }
+}
+
+list.yaml = {
+  install_info = {
+    url = "https://github.com/ikatyang/tree-sitter-yaml",
+    files = { "src/parser.c", "src/scanner.cc" },
+  }
+}
+
+list.nix = {
+  install_info = {
+    url = "https://github.com/cstrahan/tree-sitter-nix",
+    files = { "src/parser.c", "src/scanner.cc" },
+  },
+  maintainers = {"@leo60228"},
+}
+
+list.dart = {
+  install_info = {
+    url = "https://github.com/UserNobody14/tree-sitter-dart",
+    files = { "src/parser.c", "src/scanner.c" },
+  },
+  maintainers = {"@Akin909"},
+}
+
+list.rst = {
+  install_info = {
+    url = "https://github.com/stsewd/tree-sitter-rst",
+    files = { "src/parser.c", "src/scanner.c" },
+  },
+  maintainers = {"@stsewd"},
+}
+
+list.fennel = {
+  install_info = {
+    url = "https://github.com/travonted/tree-sitter-fennel",
+    files = { "src/parser.c", "src/scanner.c" },
+  },
+  maintainers = {'@TravonteD'},
+}
+
+list.teal = {
+  install_info = {
+    url = "https://github.com/euclidianAce/tree-sitter-teal",
+    files = { "src/parser.c", "src/scanner.c" },
+  },
+  maintainers = {'@euclidianAce'},
+}
+
+list.ql = {
+  install_info = {
+    url = "https://github.com/tree-sitter/tree-sitter-ql",
+    files = { "src/parser.c" },
+  },
+  maintainers = {'@pwntester'},
+}
+
+list.verilog = {
+  install_info = {
+    url = "https://github.com/tree-sitter/tree-sitter-verilog",
+    files = { "src/parser.c" },
+  },
+  used_by = { "systemverilog" },
+  maintainers = { "@zegervdv" },
+}
+-- Parsers for injections
+list.regex = {
+  install_info = {
+    url = "https://github.com/tree-sitter/tree-sitter-regex",
+    files = { "src/parser.c" }
+  },
+  maintainers = {"@theHamsta"},
+}
+
+list.jsdoc = {
+  install_info = {
+    url = "https://github.com/tree-sitter/tree-sitter-jsdoc",
+    files = { "src/parser.c" }
+  },
+  maintainers = {"@steelsojka"},
+}
+
+list.query = {
+  install_info = {
+    url = "https://github.com/nvim-treesitter/tree-sitter-query",
+    files = { "src/parser.c" }
+  },
+  readme_name = "Tree-sitter query language",
+  maintainers = {"@steelsojka"},
+}
+
+local M = {
+  list = list
+}
+
+function M.get_parser_configs()
+  return M.list
+end
+
+function M.add(parser, options)
+  M.list[parser] = options
+end
+
+function M.use_by(parser, filetype)
+  if (M.list[parser] ~= nil) then
+    local old_used_by = M.list[parser].used_by or {}
+    table.insert(old_used_by, filetype)
+    M.list[parser].used_by = old_used_by
+                print('updated the config')
+  end
```

Comment:
This can be implemented by a meta table. It could also update the config after the call of setup. Can't we just remove those three methods when the table is now public?

Reply:
Maybe the use_by can be a shortcut for newbies. But then it could also invalidate the filetype to parser cache and not print anything.

Reply:
Sorry, I thought I removed all print line code.

About meta table I was thinking the same, I will work soon on that

Reply:
I think a better solution would be if you function would just set the mapping in `ft_to_parsername` in parser.lua. This also works after setup.

Reply:
@theHamsta I think I simplified the changes with metatables and more clear interface to configure. Waiting for your feedback

I cleaned history as well to avoid many commits.

Reply:
Hi. Sorry for all the efforts. What I actually meant to say was that you could just do 

```
function M.set_parser_for_filetype(ft, parser_lang) 
     ft_to_parsername[ft] = lang
end
```
because most people like you want to set the parser for their filetype and not vice-versa. I came with this solution very late.

But your solution is maybe a bit more consistent (only one way to configure things).